Wrappers/Python/test/test_approximate_gradient.py

```diff
@@ -200,4 +201,310 @@ def test_SGD_toy_example(self):
             alg_stochastic.x.as_array(), b.as_array(), 3)
 
 
-    
\ No newline at end of file
+
+class TestSAG(CCPiTestClass): 
+
+    
+    def setUp(self):
+
+           
+        self.sampler = Sampler.random_with_replacement(6)
+        self.initial = VectorData(np.ones(30))
+        b = VectorData(np.array(range(30))/50)
+        self.n_subsets = 6
+        self.f_subsets = []
+        for i in range(6):
+            diagonal = np.zeros(30)
+            diagonal[5*i:5*(i+1)] = 1
+            Ai = MatrixOperator(np.diag(diagonal))
+            self.f_subsets.append(LeastSquares(Ai, Ai.direct(b)))
+        self.A=MatrixOperator(np.diag(np.ones(30)))
+        self.f = LeastSquares(self.A, b)
+        self.f_stochastic = SAGFunction(self.f_subsets, self.sampler)
+
+    
+    def test_approximate_gradient(self): #Test when we the approximate gradient is not equal to the full gradient 
+        with self.assertRaises(AssertionError):
+            self.assertNumpyArrayEqual(self.f_stochastic.full_gradient(self.initial).as_array(),self.f_stochastic.gradient(self.initial).as_array())
```

Comment:
This looks entirely the same as the test above in `TestSGD`.
Can you make `TestSAG(TestSGD)`?

Reply:
Done - I think